### ISSUE TITLE
Suggestion [`no-named-default`]: add option to allow type imports with multiple imports

### ISSUE THREAD

**Issue by cherryblossom000** [benmosher/eslint-plugin-import#1869]
Typescript's [`import type`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) cannot be used to import the default export and named exports without using `default as`:

```ts
import type Foo, {Bar} from 'foo'
// A type-only import can specify a default import or named bindings, but not both.
```

However, these both work but trigger `no-duplicates` and `no-named default` respectively:

```ts
import type Foo from 'foo'
import type {Bar} from 'foo'
```

```ts
import type {default as Foo, Bar} from 'foo'
```

Could there be a new option to `no-named-default` that allows type imports with multiple imports? For example:

```ts
import type {default as Foo} from 'foo' // lint warning
import type {default as Foo, Bar} from 'foo' // fine
```

**Comment by ljharb** [comment]
That seems like a bug in TypeScript; `import type` should surely be able to specify both. Has one been filed upstream?

**Comment by cherryblossom000** [comment]
@ljharb It's not a bug. From [the handbook](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export):

> If you’ve used Flow before, the syntax is fairly similar. One difference is that we’ve added a few restrictions to avoid code that might appear ambiguous.
>
> ```ts
> // Is only 'Foo' a type? Or every declaration in the import?
> // We just give an error because it's not clear.
>
> import type Foo, { Bar, Baz } from "some-module";
> //     ~~~~~~~~~~~~~~~~~~~~~~
> // error! A type-only import can specify a default import or named bindings, but not both.
> ```

**Comment by ljharb** [comment]
While I find that ridiculous, that certainly means `no-duplicate` should (by default) allow:
```jsx
import type Foo from 'foo'
import type {Bar} from 'foo'
```

**Comment by seahindeniz** [comment]
This one is really a bummer.

However, I'm using this approach to avoid getting complains
```ts
import type * as FooTypes from "./RadioSection";

const main: FooTypes.default;
const props: FooTypes.FooPropTypes;
```
I know, it's an ugly one but it's a workaround. 

I would like to give 👍 to @cherryblossom000 's proposal
```ts
import type {default as Foo, Bar} from 'foo'
```

**Comment by Riyadevvarshney11** [comment]
I would like to work on this issue

**Comment by ljharb** [comment]
Go for it!
